src/client/serialize/daos_serialize.c

```diff
@@ -206,42 +191,43 @@ serialize_uint(hid_t file_id, uint64_t val, const char *prop_str)
 static char*
 prop_to_str(uint32_t type)
 {
-	if (type == DAOS_PROP_CO_LABEL) {
-		return "DAOS_PROP_CO_LABEL";
-	} else if (type == DAOS_PROP_CO_OWNER) {
-		return "DAOS_PROP_CO_OWNER";
-	} else if (type == DAOS_PROP_CO_OWNER_GROUP) {
-		return "DAOS_PROP_CO_OWNER_GROUP";
-	} else if (type == DAOS_PROP_CO_ACL) {
-		return "DAOS_PROP_CO_ACL";
-	} else if (type == DAOS_PROP_CO_LAYOUT_TYPE) {
-		return "DAOS_PROP_CO_LAYOUT_TYPE";
-	} else if (type == DAOS_PROP_CO_LAYOUT_VER) {
-		return "DAOS_PROP_CO_LAYOUT_VER";
-	} else if (type == DAOS_PROP_CO_CSUM) {
-		return "DAOS_PROP_CO_CSUM";
-	} else if (type == DAOS_PROP_CO_CSUM_CHUNK_SIZE) {
-		return "DAOS_PROP_CO_CSUM_CHUNK_SIZE";
-	} else if (type == DAOS_PROP_CO_CSUM_SERVER_VERIFY) {
-		return "DAOS_PROP_CO_CSUM_SERVER_VERIFY";
-	} else if (type == DAOS_PROP_CO_REDUN_FAC) {
-		return "DAOS_PROP_CO_REDUN_FAC";
-	} else if (type == DAOS_PROP_CO_REDUN_LVL) {
-		return "DAOS_PROP_CO_REDUN_LVL";
-	} else if (type == DAOS_PROP_CO_SNAPSHOT_MAX) {
-		return "DAOS_PROP_CO_SNAPSHOT_MAX";
-	} else if (type == DAOS_PROP_CO_COMPRESS) {
-		return "DAOS_PROP_CO_COMPRESS";
-	} else if (type == DAOS_PROP_CO_ENCRYPT) {
-		return "DAOS_PROP_CO_ENCRYPT";
-	} else if (type == DAOS_PROP_CO_DEDUP) {
-		return "DAOS_PROP_CO_DEDUP";
-	} else if (type == DAOS_PROP_CO_DEDUP_THRESHOLD) {
-		return "DAOS_PROP_CO_DEDUP_THRESHOLD";
-	} else if (type == DAOS_PROP_CO_ALLOCED_OID) {
-		return "DAOS_PROP_CO_ALLOCED_OID";
-	} else {
-		return "PROP NOT SUPPORTED";
+	switch (type) {
+		case DAOS_PROP_CO_LABEL:
+			return "DAOS_PROP_CO_LABEL";
+		case DAOS_PROP_CO_OWNER:
+			return "DAOS_PROP_CO_OWNER";
+		case DAOS_PROP_CO_OWNER_GROUP:
+			return "DAOS_PROP_CO_OWNER_GROUP";
+		case DAOS_PROP_CO_ACL:
+			return "DAOS_PROP_CO_ACL";
+		case DAOS_PROP_CO_LAYOUT_TYPE:
+			return "DAOS_PROP_CO_LAYOUT_TYPE";
+		case DAOS_PROP_CO_LAYOUT_VER:
+			return "DAOS_PROP_CO_LAYOUT_VER";
+		case DAOS_PROP_CO_CSUM:
+			return "DAOS_PROP_CO_CSUM";
+		case DAOS_PROP_CO_CSUM_CHUNK_SIZE:
+			return "DAOS_PROP_CO_CSUM_CHUNK_SIZE";
+		case DAOS_PROP_CO_CSUM_SERVER_VERIFY:
+			return "DAOS_PROP_CO_CSUM_SERVER_VERIFY";
+		case DAOS_PROP_CO_REDUN_FAC:
+			return "DAOS_PROP_CO_REDUN_FAC";
+		case DAOS_PROP_CO_REDUN_LVL:
+			return "DAOS_PROP_CO_REDUN_LVL";
+		case DAOS_PROP_CO_SNAPSHOT_MAX:
+			return "DAOS_PROP_CO_SNAPSHOT_MAX";
+		case DAOS_PROP_CO_COMPRESS:
+			return "DAOS_PROP_CO_COMPRESS";
+		case DAOS_PROP_CO_ENCRYPT:
+			return "DAOS_PROP_CO_ENCRYPT";
+		case DAOS_PROP_CO_DEDUP:
+			return "DAOS_PROP_CO_DEDUP";
+		case DAOS_PROP_CO_DEDUP_THRESHOLD:
+			return "DAOS_PROP_CO_DEDUP_THRESHOLD";
+		case DAOS_PROP_CO_ALLOCED_OID:
+			return "DAOS_PROP_CO_ALLOCED_OID";
+		default:	
+			return "PROPERTY NOT SUPPORTED";
```

Comment:
I'm not sure what the best default here should be. What distinguishes this text from an actual property name? Maybe this should return NULL and the caller is responsible for checking this?

Reply:
@daltonbohning I am not sure either. If it does go to the default it will not match any types in the serialize_props function though, and just return that serialization of that property type is not supported with a DER_INVAL error code. It is distinguished because serialize_props checks the list of prop types it supports.

Reply:
@dsikich I guess because this is a private/static function it's okay, but if we ever make this function public (such as moving to an internal DAOS header) then we could figure out how to best handle this.

Reply:
@daltonbohning yeah, that makes sense.